spec/requests/atmosphere/api/actions_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+describe Atmosphere::Api::V1::ActionsController do
+  include ApiHelpers
+
+  describe 'GET /actions' do
+
+    let(:optimizer) { double('optimizer') }
+
+    before do
+      expect(Atmosphere::Optimizer).to receive(:instance).at_least(:once) { optimizer }
+      expect(optimizer).to receive(:run).at_least(:once).with(anything())
+    end
+
+    let!(:message) { "message123" }
+
+    let!(:user) { create(:user) }
+    let!(:appliance_set)  { create(:appliance_set, :user => user) }
+    let!(:appliance)  { create(:appliance, :appliance_set => appliance_set) }
+
+
+    let!(:user2) { create(:user) }
+    let!(:appliance_set2)  { create(:appliance_set, :user => user2) }
+    let!(:appliance2)  { create(:appliance, :appliance_set => appliance_set2) }
+
+    let!(:action_log) { create(:action_log, :message => message) }
+    let!(:action)  { create(:action, :appliance => appliance, :action_logs => [action_log]) }
+    let!(:other_action)  { create(:action, :appliance => appliance2) }
```

Comment:
You know that `let!` is creating object in DB before every spec - even if it is not used in concrete test case. It is better to create only minimum set of DB objects which are really needed for the spec.